### ISSUE TITLE
[RFE] Allow selection of credential on the workflow level.

### ISSUE THREAD

**Issue by saali2016** [ansible/awx#7859]
##### ISSUE TYPE
Can we have an option on workflow "prompt on launch" for credential

##### SUMMARY
This would be similar to the following issue and pull, except it would be for workflow:

https://github.com/ansible/awx/issues/2256
https://github.com/ansible/awx/pull/2342

**Comment by stahloss** [comment]
Any news on this? We would be helped so much by this feature.

**Comment by Klistel** [comment]
Adding a +1 on this feature request. I have workflows that I use prompts to select inventories and hosts for my job templates, and we have SSH keys for "prod" and "nonprod" - not being able to select for credential at workflow launch time means I have to have two distinct workflows for the same procedure, one for prod and one for nonprod. This would be very useful to me - I'm not sure why prompting for credentials is allowed at the Template level but not Workflow level

**Comment by rroethof** [comment]
Also adding +1 here, currently we can not use a singe workflow for multiple inventories (OTAP/DTAP)

**Comment by cdorsman** [comment]
Also adding +1

**Comment by anthoc** [comment]
Also adding +1

**Comment by ygiri1** [comment]
+1 it would help alot

**Comment by jamesregis** [comment]
Also adding +1

**Comment by rezaa1** [comment]
Would be great feature +1
 We need to run workflow with the individual user and password due to security reason, and can't get that working,

**Comment by chiquiff** [comment]
+1

**Comment by kTipSSIoYv** [comment]
Also adding +1

**Comment by WoodooTek** [comment]
\+ 1

**Comment by testotxt** [comment]
+1

**Comment by dolgovas** [comment]
+1

**Comment by RemonIbrahimNashed** [comment]
+1

**Comment by hvdkooij** [comment]
Count me in. Having this option would allow us to do more jobs which we can't use right now.

**Comment by mgomon-usc** [comment]
Yeah same here. Kinda figured this would have been done by now :D

**Comment by arnaud-asp** [comment]
+1

**Comment by borucltd** [comment]
+1 that option would be very useful

**Comment by bherrin3** [comment]
+1

**Comment by jon-nfc** [comment]
+1

**Comment by bherrin3** [comment]
I am going to +1 this again as I just had a workflow failure in the context of `ovirt_auth` RHV credential caching which does not allow two separate RHVs to use the same workflow since the credential cannot be `prompt on launch`

**Comment by taazich1** [comment]
+1

**Comment by usunnapu** [comment]
+1. Without this, it trumps all the advantages of workflow template

**Comment by AlekseiKulberg** [comment]
+1

**Comment by Mazadon** [comment]
+1

**Comment by azrdev** [comment]
Requested for AAP as https://issues.redhat.com/browse/AAPRFE-199

**Comment by Stavros** [comment]
+1

**Comment by giulianoquites** [comment]
+1

**Comment by lepesh14** [comment]
+1

**Comment by joeashcraft** [comment]
+1 

---
as a developer of job templates and workflows in AWX/AAP, I want to create a single job-template for use in a single workflow and have the job-template use the appropriate credential from a list of 2 Machine Credentials.

today:
* a job-template can only associate with 1 Machine Credential
* I have 2 Hosts in an Inventory, and each Host can only be accessed by the right Machine Credential
* I have 2 Machine Credentials with username and password defined

today's workaround: 
* I create multiple job-templates that are nearly identical except for the Machine Credential selected
